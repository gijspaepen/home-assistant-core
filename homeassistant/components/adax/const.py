"""Constants for the Adax integration."""

from typing import Final

from homeassistant.const import Platform

ACCOUNT_ID: Final = "account_id"
CLOUD = "Cloud"
CONNECTION_TYPE = "connection_type"
DOMAIN: Final = "adax"
LOCAL = "Local"
WIFI_SSID = "wifi_ssid"
WIFI_PSWD = "wifi_pswd"

PLATFORMS = [Platform.CLIMATE, Platform.SENSOR]
