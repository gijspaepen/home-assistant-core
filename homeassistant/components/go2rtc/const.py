"""Go2rtc constants."""

DOMAIN = "go2rtc"

CONF_DEBUG_UI = "debug_ui"
EXCLUSE_ERROR_MESSAGE = "Url and debug_ui cannot be set at the same time."
