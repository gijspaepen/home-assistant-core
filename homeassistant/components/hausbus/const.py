"""Constants for the Haus-Bus integration."""

from __future__ import annotations

import logging

LOGGER = logging.getLogger(__package__)

DOMAIN = "hausbus"
ATTR_ON_STATE = "on_state"
