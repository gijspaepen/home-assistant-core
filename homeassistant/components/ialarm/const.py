"""Constants for the iAlarm integration."""

from pyialarm import IAlarm

from homeassistant.components.alarm_control_panel import AlarmControlPanelEntityState

DATA_COORDINATOR = "ialarm"

DEFAULT_PORT = 18034

DOMAIN = "ialarm"

IALARM_TO_HASS = {
    IAlarm.ARMED_AWAY: AlarmControlPanelEntityState.ARMED_AWAY,
    IAlarm.ARMED_STAY: AlarmControlPanelEntityState.ARMED_HOME,
    IAlarm.DISARMED: AlarmControlPanelEntityState.DISARMED,
    IAlarm.TRIGGERED: AlarmControlPanelEntityState.TRIGGERED,
}
