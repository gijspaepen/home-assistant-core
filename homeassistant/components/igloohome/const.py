"""Constants for the igloohome integration."""

DOMAIN = "igloohome"

OAUTH2_AUTHORIZE = "https://auth.igloohome.co/login"
OAUTH2_TOKEN = "https://auth.igloohome.co/oauth2/token"
