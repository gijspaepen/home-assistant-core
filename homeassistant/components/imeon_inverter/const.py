"""Constant for Imeon component."""

DOMAIN = "imeon_inverter"
HUBNAME = "imeon_inverter_hub"
TIMEOUT = 10  # seconds
