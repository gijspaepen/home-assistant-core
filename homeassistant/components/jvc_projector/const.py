"""Constants for the jvc_projector integration."""

from jvcprojector import const

NAME = "JVC Projector"
DOMAIN = "jvc_projector"
MANUFACTURER = "JVC"

REMOTE_COMMANDS = {
    "menu": const.REMOTE_MENU,
    "up": const.REMOTE_UP,
    "down": const.REMOTE_DOWN,
    "left": const.REMOTE_LEFT,
    "right": const.REMOTE_RIGHT,
    "ok": const.REMOTE_OK,
    "back": const.REMOTE_BACK,
    "mpc": const.REMOTE_MPC,
    "hide": const.REMOTE_HIDE,
    "info": const.REMOTE_INFO,
    "input": const.REMOTE_INPUT,
    "cmd": const.REMOTE_CMD,
    "advanced_menu": const.REMOTE_ADVANCED_MENU,
    "picture_mode": const.REMOTE_PICTURE_MODE,
    "color_profile": const.REMOTE_COLOR_PROFILE,
    "lens_control": const.REMOTE_LENS_CONTROL,
    "setting_memory": const.REMOTE_SETTING_MEMORY,
    "gamma_settings": const.REMOTE_GAMMA_SETTINGS,
    "hdmi_1": const.REMOTE_HDMI_1,
    "hdmi_2": const.REMOTE_HDMI_2,
    "mode_1": const.REMOTE_MODE_1,
    "mode_2": const.REMOTE_MODE_2,
    "mode_3": const.REMOTE_MODE_3,
    "lens_ap": const.REMOTE_LENS_AP,
    "gamma": const.REMOTE_GAMMA,
    "color_temp": const.REMOTE_COLOR_TEMP,
    "natural": const.REMOTE_NATURAL,
    "cinema": const.REMOTE_CINEMA,
    "anamo": const.REMOTE_ANAMO,
    "3d_format": const.REMOTE_3D_FORMAT,
}
