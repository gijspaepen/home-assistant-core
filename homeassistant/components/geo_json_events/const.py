"""Define constants for the GeoJSON events integration."""

from __future__ import annotations

from typing import Final

from homeassistant.const import Platform

DOMAIN: Final = "geo_json_events"

PLATFORMS: Final = [Platform.GEO_LOCATION]

ATTR_EXTERNAL_ID: Final = "external_id"
DEFAULT_RADIUS_IN_KM: Final = 20.0
DEFAULT_RADIUS_IN_M: Final = 20000.0
DEFAULT_UPDATE_INTERVAL: Final = 300
SOURCE: Final = "geo_json_events"
CONF_UPDATE_INTERVAL: Final = "update_interval"

SIGNAL_DELETE_ENTITY: Final = "geo_json_events_delete_{}"
SIGNAL_UPDATE_ENTITY: Final = "geo_json_events_update_{}"
