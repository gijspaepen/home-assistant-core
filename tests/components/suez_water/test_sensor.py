"""Test init of Suez water component."""

from unittest.mock import AsyncMock, patch

from syrupy import SnapshotAssertion

from homeassistant.components.suez_water import PySuezError, SuezClient
from homeassistant.config_entries import ConfigEntryState
from homeassistant.const import Platform
from homeassistant.core import HomeAssistant
from homeassistant.helpers import entity_registry as er

from . import setup_integration

from tests.common import MockConfigEntry


async def test_init_success(
    suez_client,
    config_entry: MockConfigEntry,
    snapshot: SnapshotAssertion,
    hass: HomeAssistant,
    entity_registry: er.EntityRegistry,
) -> None:
    """Test that the entry is loaded and has valid state."""
    with patch("homeassistant.components.suez_water.PLATFORMS", [Platform.SENSOR]):
        await setup_integration(suez_client, config_entry, hass)

        assert config_entry.state is ConfigEntryState.LOADED
        entity_entries = er.async_entries_for_config_entry(
            entity_registry, config_entry.entry_id
        )

        assert entity_entries
        assert len(entity_entries) == 1
        for entity_entry in entity_entries:
            assert entity_entry == snapshot(name=f"{entity_entry.entity_id}-entry")
            assert hass.states.get(entity_entry.entity_id) == snapshot(
                name=f"{entity_entry.entity_id}-state"
            )


async def test_update_failed(
    config_entry: MockConfigEntry, hass: HomeAssistant
) -> None:
    """Test that the state of the entity reflect failure."""

    mock = AsyncMock(spec=SuezClient)
    mock.update.side_effect = PySuezError("Should fail to update")
    mock.check_credentials.return_value = True
    with (
        patch("homeassistant.components.suez_water.PLATFORMS", [Platform.SENSOR]),
        patch(
            "homeassistant.components.suez_water.SuezClient", return_value=mock
        ) as mock_client,
    ):
        await setup_integration(mock_client, config_entry, hass)

        assert config_entry.state is ConfigEntryState.SETUP_RETRY

        state = hass.states.get("sensor.suez_mock_device_water_usage_yesterday")
        assert state is None
