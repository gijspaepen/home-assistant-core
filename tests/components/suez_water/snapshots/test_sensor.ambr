# serializer version: 1
# name: test_suez_sensors_valid_state[sensor.suez_mock_device_water_usage_yesterday_historical_sensor-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': None,
    'entity_id': 'sensor.suez_mock_device_water_usage_yesterday_historical_sensor',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.WATER: 'water'>,
    'original_icon': None,
    'original_name': 'Water usage yesterday - historical sensor',
    'platform': 'suez_water',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'water_usage_yesterday',
    'unique_id': 'test-counter_water_usage_yesterday',
    'unit_of_measurement': <UnitOfVolume.LITERS: 'L'>,
  })
# ---
# name: test_suez_sensors_valid_state[sensor.suez_mock_device_water_usage_yesterday_historical_sensor-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'attribution': 'suez water mock test',
      'device_class': 'water',
      'friendly_name': 'Suez mock device Water usage yesterday - historical sensor',
      'highest_monthly_consumption': 2558,
      'history': dict({
        '2024-01-01': 130,
        '2024-01-02': 145,
        '2024-12-01': 154,
        '2024-12-02': 166,
      }),
      'last_year_overall': 1000,
      'previous_month_consumption': dict({
        '2024-12-01': 154,
        '2024-12-02': 166,
      }),
      'this_month_consumption': dict({
        '2024-01-01': 130,
        '2024-01-02': 145,
      }),
      'this_year_overall': 1500,
      'unit_of_measurement': <UnitOfVolume.LITERS: 'L'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.suez_mock_device_water_usage_yesterday_historical_sensor',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '160',
  })
# ---
